api/controllers/slackbotController.js

```diff
@@ -0,0 +1,242 @@
+      },
+    };
+  }
+
+  if (!regexpURL.test(cover)) {
+    return {
+      response_action: "errors",
+      errors: {
+        cover: "You must enter a valid URL",
+      },
+    };
+  }
+
+  return "";
+};
+
+const validateRegisterForm = (values) => {
+  const firstName = values.first_name.first_name.value;
+  const lastName = values.last_name.last_name.value;
+  const username = values.username;
+  const email = values.email.email.value;
+  const password = values.password.password.value;
+  regexEMAIL = /^\S+@\S+$/gm;
+
+  if (!firstName || !lastName || !password || !username || !email) {
+    return {
+      response_action: "errors",
+      errors: {
+        channel: "You must fill in all the fields",
+      },
+    };
+  }
+
+  if (!regexEMAIL.test(email)) {
+    return {
+      response_action: "errors",
+      errors: {
+        email: "You must enter a valid Email",
+      },
+    };
+  }
+
+  return "";
+};
+
+Bot.handleInteractions = async (req, res, next) => {
+  const payload = JSON.parse(req.body.payload);
+  const { values } = payload.view.state;
+  const userInfo = await web.users.info({
+    user: payload.user.id,
+  });
+  responseURL = payload.response_urls[0].response_url;
+  values.username = userInfo.user.real_name;
+  let validateResponse;
+
+  if (payload.type === "view_submission") {
+    switch (payload.view.callback_id) {
+      case "add-book":
+        validateResponse = validateBookForm(values);
+        res.status(200).send(validateResponse);
+        if (validateResponse === "") {
+          addBook(values, req, res, next);
+        }
+        break;
+      case "add-user":
+        validateResponse = validateRegisterForm(values);
+        res.status(200).send(validateResponse);
+        if (validateResponse === "") {
+          addUser(values, req, res, next);
+        }
+        break;
+      default:
+        console.log("Unknown interaction");
+        return res.status(500).send("Unknown interaction");
+    }
+  }
+};
+
+const addBook = async (values, req, res, next) => {
+  try {
+    const title = values.title.title.value;
+    const author = values.author.author.value;
+    const cover = values.cover.cover.value;
+    const abstract = values.abstract.abstract.value;
+    const publication_date =
+      values.publication_date.publication_date.selected_date;
+    const user = values.username;
+
+    const findedUser = await userModel.findOne({
+      where: { username: user },
+    });
+
+    const book = await bookModel.create({
+      title,
+      author,
+      publication_date,
+      abstract,
+      cover,
+      userId: findedUser.id,
+    });
+    let responseMessage = await fetch(responseURL, {
+      method: "post",
+      body: JSON.stringify({
+        text: "Book added successfully! :partying_face:",
+        response_type: "ephemeral",
+      }),
+      headers: { "Content-Type": "application/json" },
+    });
+    console.log(responseMessage);
```

Comment:
Well, Do you really need this console.log()?

Reply:
No i don't. Thanks for the observation :raised_hands: